src/core/connections.jl

```diff
@@ -278,94 +278,152 @@ function split_datum_path(obj::AbstractCompositeComponentDef, s::AbstractString)
     return (ComponentPath(obj, elts[1]), Symbol(elts[2]))
 end
 
-# TBD: Deprecated?
 """
-Connect a parameter and variable using string notation "/path/to/component:datum_name" where
-the potion before the ":" is the string representation of a component path from `obj` and the
-portion after is the name of the src or dst datum.
+    _find_paths_and_names(obj::AbstractComponentDef, datum_name::Symbol)
+
+Recurses through sub components and finds the full path(s) to desired datum, and their
+names at the leaf level. Returns a tuple (paths::Vector{ComponentPath}, datum_names::Vector{Symbol})
 """
-function connect_param!(obj::AbstractCompositeComponentDef, dst::AbstractString, src::AbstractString,
-                        backup::Union{Nothing, Array}=nothing; ignoreunits::Bool=false, offset::Int=0)
-    dst_path, dst_name = split_datum_path(obj, dst)
-    src_path, src_name = split_datum_path(obj, src)
+function _find_paths_and_names(obj::AbstractComponentDef, datum_name::Symbol)
+
+    # Base case-- leaf component
+    if obj isa ComponentDef
+        return ([nothing], [datum_name])
+    end
```

Comment:
note to self: might clean up this base case to actually return the comp name instead of nothing, after the comp_path bug is fixed separately